### PR TITLE
Adding ingest validation implementation

### PR DESCRIPTION
**Pull request by pdelewski**
This PR adds validation to data ingest.
If a json field has a type that is different from db column, it will be added to attributes.

### CODE REVIEW

File: quesma/clickhouse/clickhouse.go [hunk 1]
```diff
@@ -421,9 +421,18 @@ func (lm *LogManager) CreateTableFromInsertQuery(ctx context.Context, name strin
 	return nil
 }
 
+// This function takes an attributesMap and update it
+// with the fields that are not validated according to the inferred schema
+func addInvalidJsonFieldsToAttributes(attrsMap map[string][]interface{}, inValidJson types.JSON) {
```
Comment (avelanarius) [inline quesma/clickhouse/clickhouse.go:426]:
```suggestion
func addInvalidJsonFieldsToAttributes(attrsMap map[string][]interface{}, invalidJson types.JSON) {
```

here and in other places. The previous capitalization made it a bit confusing for me (it suggested "inValidJson = in valid JSON").

File: quesma/clickhouse/clickhouse.go [hunk 2]
```diff
@@ -421,9 +421,18 @@ func (lm *LogManager) CreateTableFromInsertQuery(ctx context.Context, name strin
 	return nil
 }
 
+// This function takes an attributesMap and update it
+// with the fields that are not validated according to the inferred schema
```
Comment (avelanarius) [inline quesma/clickhouse/clickhouse.go:425]:
```suggestion
// This function takes an attributesMap and updates it
// with the fields that are not valid according to the inferred schema
```

File: quesma/clickhouse/clickhouse.go [hunk 3]
```diff
@@ -473,6 +481,10 @@ func (lm *LogManager) BuildInsertJson(tableName string, data types.JSON, config
 	} else {
 		return "", fmt.Errorf("no attributes or others in config, but received non-schema fields: %s", mDiff)
 	}
+	// If there are some non-valid fields, we need to add them to the attributes map
```
Comment (avelanarius) [inline quesma/clickhouse/clickhouse.go:484]:
```suggestion
	// If there are some invalid fields, we need to add them to the attributes map
```

File: quesma/clickhouse/ingest_validator.go [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+	incomingValueType := getTypeName(value)
+	if columnType == DateTimeType {
+		// TODO validate date format
+		// For now we store dates as strings
+		if incomingValueType != StringType {
+			deletedFields[fieldName] = value
+		}
+	} else if columnType != incomingValueType {
+		deletedFields[fieldName] = value
+	}
```
Comment (avelanarius) [inline quesma/clickhouse/ingest_validator.go:91]: This should also allow inserting ints as floats (in one ingest data we examined, a user sometimes inserted "int" `0` to a float column)

**Review by avelanarius — COMMENTED** [review]
Looks OK apart from the comments I left.

**Review by avelanarius — APPROVED** [review]

File: quesma/clickhouse/ingest_validator.go (hunk 4 above)
Comment (pdelewski) [reply quesma/clickhouse/ingest_validator.go:91]: Agree.

**Pull request merged**